src/test/java/htsjdk/samtools/util/IoUtilTest.java

```diff
@@ -187,4 +191,54 @@ public void testFileTypeUnix(final String path, boolean expectedIsRegularFile) {
                 {"/non/existent/file", Boolean.TRUE},
         };
     }
+    
+    @Test
+    public void isGZIPInputStreamTest() throws IOException {

```

Comment:
please use a dataprovider here too instead of testing two different scenarios in a single test.